extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProviderPlugin.java

```diff
@@ -33,6 +35,13 @@ public SeekableChannelsProvider createProvider(@NotNull final URI uri, @Nullable
             throw new IllegalArgumentException("Arguments not compatible, provided uri " + uri);
         }
         final S3Instructions s3Instructions = (S3Instructions) config;
-        return new S3SeekableChannelProvider(s3Instructions);
+        if (instance == null || s3Instructions != instance.getS3Instructions()) {
+            synchronized (S3SeekableChannelProvider.class) {
+                if (instance == null || !s3Instructions.equals(instance.getS3Instructions())) {
+                    instance = new S3SeekableChannelProvider(s3Instructions);
+                }
+            }
+        }
+        return instance;
```

Comment:
This is racy given the volatile reads. You'll either need to be careful w/ your volatile reads into a local variable, use a cache/map instead, or revert.